### PR TITLE
[DEN-228] Create Onboarding.md

### PR DESCRIPTION
**Pull request by stevenpaz**

### CODE REVIEW

File: Onboarding.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Onboarding: ec-go-sdk
+
+## Purpose
+This document is a collection of all onboarding related information, tips & 
+tricks, etc. for first-time SDK contributers.
```
Comment (defstream) [inline Onboarding.md:5]: This document is a collection of all `onboarding-related` information, tips & 
tricks, etc. for first-time SDK `contributors`.

File: Onboarding.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    - Sandbox for playing with Go code.
+- [Go Docs](https://golang.org/doc/)
+    - Landing page for many Go resources e.g. learning, concepts, etc.
+- [Effective Go](https://go.dev/doc/effective_go)
+    - Outlines community-accepted code style and best practices.
+- [Our Go Guide](https://github.com/EdgeCast/ec-sdk-go/blob/main/Go.md)
+    - Best practices and code style our team adheres to.
+
+## Local Environment Setup
+If you have a Pluralsight license, we recommend you watch the Getting Started
```
Comment (defstream) [inline Onboarding.md:27]: If you have a Pluralsight license, we recommend `watching` the Getting Started

File: Onboarding.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+[Architecture.md](Architecture.md) files.
+
+## Development Workflow
+Create a branch off of main and begin coding! 
+
+### Default Environment Configuration
+Please note that the SDK is configured to point to the EdgeCast production 
+environment. This is fine if you own a test account. Developers employed at 
+EdgeCast may wish to point to a different environment. You can do so globally by 
+modifying the URLs in [config.go](edgecast/config.go) or at runtime by setting
```
Comment (defstream) [inline Onboarding.md:54]: or `during` runtime

**Review by defstream — CHANGES_REQUESTED** [review]

File: Onboarding.md (hunk 1 above)
Comment (stevenpaz) [reply Onboarding.md]: Fixed

File: Onboarding.md (hunk 2 above)
Comment (stevenpaz) [reply Onboarding.md]: Fixed

File: Onboarding.md (hunk 3 above)
Comment (stevenpaz) [reply Onboarding.md]: Fixed

**Review by defstream — APPROVED** [review]

**Pull request merged**